src/context.c

```diff
@@ -427,8 +377,12 @@ static gboolean r_context_configure_target(GError **error)
 		context->bootslot = get_cmdline_bootname();
 	}
 
-	if (context->bootslot == NULL && (g_strcmp0(context->config->system_bootloader, "custom") == 0)) {
-		context->bootslot = get_custom_bootname();
+	if (context->bootslot == NULL) {
+		context->bootslot = r_boot_get_bootname(context->config, &ierror);
+		if (ierror) {
+			g_propagate_error(error, ierror);
+			return FALSE;
+		}
```

Comment:
Aborting on an error here is a behavior change and so should be explained in the commit message. Perhaps it should be a separate commit?